2023-08-22 12:17:14,963 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 12:59:40,130 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:12:10,073 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:13:15,266 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:13:37,265 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:14:42,239 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:14:51,208 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:14:57,593 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-25 13:15:40,829 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-29 11:59:46,230 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-29 13:15:06,404 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-29 13:18:37,497 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-29 13:22:29,141 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-08-29 13:48:07,049 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:27]
2023-09-01 12:30:13,216 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-01 12:31:51,760 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-01 16:36:02,611 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-01 16:46:01,767 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-01 16:53:19,718 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-01 16:54:46,396 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-04 12:25:42,867 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:29]
2023-09-04 12:35:10,851 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:30]
2023-09-04 13:04:47,142 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:30]
2023-09-05 11:38:05,010 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:30]
2023-09-05 11:38:16,182 ERROR: Exception on /login [GET] [in C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py:1414]
Traceback (most recent call last):
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\app\routes.py", line 45, in login
    return render_template('login.html', title='Login', form=form)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\templating.py", line 151, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\templating.py", line 132, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\jinja2\environment.py", line 1301, in render
    self.environment.handle_exception()
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\jinja2\environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Yash\PycharmProjects\microblog\app\templates\login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\Yash\PycharmProjects\microblog\app\templates\base.html", line 32, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\app\templates\login.html", line 27, in block 'content'
    <a href="{{ url_for('reset_password_request') }}">Click to Reset It</a>
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1697, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1686, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\werkzeug\routing\map.py", line 950, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'reset_password_request'. Did you mean 'edit_profile' instead?
2023-09-05 11:55:09,168 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:30]
2023-09-05 13:19:52,001 INFO: Microblog startup [in C:\Users\Yash\PycharmProjects\microblog\app\__init__.py:30]
2023-09-05 13:20:24,248 ERROR: Exception on /reset_password_request [POST] [in C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py:1414]
Traceback (most recent call last):
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 492, in send
    message.send(connection)
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 427, in send
    connection.send(self)
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 181, in send
    if message.has_bad_headers():
       ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 400, in has_bad_headers
    headers = [self.sender, self.reply_to] + self.recipients
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
TypeError: can only concatenate list (not "str") to list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask\app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\PycharmProjects\microblog\app\routes.py", line 176, in reset_password_request
    send_password_reset_email(user)
  File "C:\Users\Yash\PycharmProjects\microblog\app\email.py", line 15, in send_password_reset_email
    send_email('[Microblog] Reset Your Password',
  File "C:\Users\Yash\PycharmProjects\microblog\app\email.py", line 10, in send_email
    mail.send(msg)
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 491, in send
    with self.connect() as connection:
  File "C:\Users\Yash\PycharmProjects\microblog\myenv\Lib\site-packages\flask_mail.py", line 152, in __exit__
    self.host.quit()
  File "C:\Users\Yash\AppData\Local\Programs\Python\Python311\Lib\smtplib.py", line 1004, in quit
    res = self.docmd("quit")
          ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Yash\AppData\Local\Programs\Python\Python311\Lib\smtplib.py", line 431, in docmd
    self.putcmd(cmd, args)
  File "C:\Users\Yash\AppData\Local\Programs\Python\Python311\Lib\smtplib.py", line 378, in putcmd
    self.send(f'{s}{CRLF}')
  File "C:\Users\Yash\AppData\Local\Programs\Python\Python311\Lib\smtplib.py", line 365, in send
    raise SMTPServerDisconnected('please run connect() first')
smtplib.SMTPServerDisconnected: please run connect() first
